### ISSUE TITLE
[Accessibility]: Search notebooks document outline keyboard navigation 

### ISSUE THREAD

**Issue by marisak** [sourcegraph/sourcegraph#35153]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/35142

### Problem description

When using keyboard navigation with "Enable Full Keyboard Access" enabled in MacOS settings, I can't navigate to the buttons in the notebook header. I get stuck in the document outline. However, this is not a problem when "Enable Full Keyboard Access" is not enabled. I'm filing this issue because I'm not confident that I'm using "Enable Full Keyboard Access" properly, so this may not be a problem for a more skilled keyboard user. 

### Expected behavior

After tabbing through all the items in the document outline, I would expect to go to the notebook icon, then the star icon, then the buttons. 

### Additional details

In actuality, after tabbing through all the items in the document outline, I occasionally can tab to the notebook icon, then the star icon. However, I get looped back into the document outline, even without using shift-tab. This blocks "I can export and copy the notebook" and to some extent "I can open and close the document outline."

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by marisak** [comment]
I'm not 100% sure that design input is needed but I've added the label just in case.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by umpox** [comment]
Followed on this page: https://k8s.sgdev.org/notebooks/Tm90ZWJvb2s6Mw==

Couldn't replicate 🤔 

It looks like "Full keyboard access" appears to have [quite a few quirks](https://mybyways.com/blog/enable-mac-full-keyboard-access-but-beware). This is not something we're directly auditing so I think we can close this unless we're confident there's an issue here. I would imagine there are different ways of navigating in this mode.
